### PR TITLE
finished my hometask.

### PR DESCRIPTION
**Pull request by MihailNenahov**
wrote factorial function and all methods for square structere.

### CODE REVIEW

File: second_task/go/src/factorial/factorial.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package main
+
+import "fmt"
+
+func factorial(given_number uint) (result uint) { //могу ли я как-то инициализировать result прямо в шапке функции?
```
Comment (burov) [inline second_task/go/src/factorial/factorial.go]: На что то кроме 0, нельзя

Comment (burov) [inline second_task/go/src/factorial/factorial.go]: In Golang preferable to use camel case for names

File: second_task/go/src/square/square.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+func (my_square Square) End() (x_result int, y_result int) {
+	return my_square.start.x + int(my_square.a), my_square.start.y - int(my_square.a)
+}
+
+func (my_square Square) Perimeter() uint {
+	return my_square.a * 4
+}
+
+func (my_square Square) Area() uint {
+	//return uint(Pow(float64(my_square.a), float64(2))) //как использовать эту функцию?
```
Comment (burov) [inline second_task/go/src/square/square.go]: return uint(math.Pow(float64(my_square.a), float64(2)))

File: second_task/go/src/square/square.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+type Point struct {
+	x, y int
+}
+
+type Square struct {
+	start Point
+	a     uint
+}
+
+func (my_square Square) End() (x_result int, y_result int) {
```
Comment (burov) [inline second_task/go/src/square/square.go]: For functions receivers short names are preferable

**Review by burov — CHANGES_REQUESTED** [review]
Please pay attention on naming

File: second_task/go/src/factorial/factorial.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+	result=1
+	if given_number == 0 {
+		return result
+	}
+	for i:=given_number ; i>1 ; i-- {
```
Comment (imatsiash-ib) [inline second_task/go/src/factorial/factorial.go]: Code is not `go fmt`-ed

File: second_task/go/src/square/square.go (hunk 3 above)
Comment (MihailNenahov) [reply second_task/go/src/square/square.go]: I don't understand meaning of this comment. Can u explain it?

Comment (burov) [reply second_task/go/src/square/square.go]: In this case my_square is function receiver name, you use too long name for function receiver name as a common practice function receiver equal first letter or some reduced word like server -> srv, request -> req, Square -> s

Comment (MihailNenahov) [reply second_task/go/src/square/square.go]: understood.

File: second_task/factorial/factorial.go [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package main
+
+import "fmt"
+
+func Factorial(GivenNumber uint) (Result uint) {
+	Result = 1
```
Comment (imatsiash-ib) [inline second_task/factorial/factorial.go]: why do you use Exported names ?

File: second_task/square/square.go [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+type Point struct {
+	x, y int
+}
+
+type Square struct {
+	start Point
+	a     uint
+}
+
+func (Sqr Square) End() (XResult int, YResult int) {
```
Comment (imatsiash-ib) [inline second_task/square/square.go]: This method should return the `Point` structure.

**Review by imatsiash-ib — APPROVED** [review]

File: second_task/square/square.go [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+func (Sqr Square) End() Point {
+	return Point{Sqr.start.x + int(Sqr.a), Sqr.start.y - int(Sqr.a)}
+}
+
+func (Sqr Square) Perimeter() uint {
+	return Sqr.a * 4
+}
+
+func (Sqr Square) Area() uint {
+	return uint(math.Pow(float64(Sqr.a), float64(2)))
```
Comment (burov) [inline second_task/square/square.go]: rules applied to function receiver name the same as a variable use ```sqr``` instead off ```Sqr```, as well don't make things harder than needed ```sqr.a * sqr.a``` looks more readable

**Review by burov — APPROVED** [review]
Overal OK, somme minor errors and awkward moments

File: second_task/square/square.go (hunk 7 above)
Comment (MihailNenahov) [reply second_task/square/square.go]: if i'll correct it, everything else will be ok?

Comment (MihailNenahov) [reply second_task/square/square.go]: короче я исправил. потом допишете, если реально есть, что исправить.

**Pull request merged**